estate/models/estate_property.py

```diff
@@ -0,0 +1,132 @@
+    _name = "estate.property"
+    _description = "Property Data Table"
+
+    name = fields.Char(string="Title", required=True)
+    description = fields.Text(string="Description")
+    postcode = fields.Char(string="Postcode")
+    date_availability = fields.Date(
+        string="Available From",
+        copy=False,
+        default=lambda self: date.today() + relativedelta(months=3),
+    )
+    expected_price = fields.Float(string="Expected Price", required=True)
+    selling_price = fields.Float(string="Selling Price", copy=False, readonly=True)
+    bedrooms = fields.Integer(string="Bedrooms", default=2)
+    living_area = fields.Integer(string="Living Area (sqm)")
+    facades = fields.Integer(string="Facades")
+    garage = fields.Boolean(string="Garage")
+    garden = fields.Boolean(string="Garden")
+    garden_area = fields.Integer(string="Garden Area (sqm)")
+    garden_orientation = fields.Selection(
+        selection=[
+            ("north", "North"),
+            ("south", "South"),
+            ("east", "East"),
+            ("west", "West"),
+        ]
+    )
+    active = fields.Boolean(string="Active", default=False)
+    state = fields.Selection(
+        string="State",
+        selection=[
+            ("new", "New"),
+            ("offer_received", "Offer Received"),
+            ("offer_accepted", "Offer Accepted"),
+            ("sold", "Sold"),
+            ("cancelled", "Cancelled"),
+        ],
+        required=True,
+        copy=False,
+        default="new",
+    )
+
+    property_type_id = fields.Many2one("estate.property.type", string="Property Type")
+    salesperson_id = fields.Many2one(
+        "res.users", string="Salesman", index=True, default=lambda self: self.env.user
+    )
+    buyer_id = fields.Many2one("res.partner", string="Buyer", index=True, copy=False)
+    tag_ids = fields.Many2many("estate.property.tag", string="Property Tags")
+    offer_ids = fields.One2many("estate.property.offer", "property_id", string="Offers")
+
+    total_area = fields.Float(compute="_compute_total_area", string="Total Area (sqm)")
+    best_price = fields.Float(compute="_compute_best_price", string="Best Offer")
+
+    _sql_constraints = [
+        (
+            "check_expected_price_positive",
+            "CHECK(expected_price > 0)",
+            "The expected price must be strictly positive.",
+        ),
+        (
+            "check_selling_price_positive",
+            "CHECK(selling_price >= 0)",
+            "The selling price must be positive.",
+        ),
+    ]
+
+    @api.constrains("expected_price", "selling_price")
+    def _check_selling_price(self):
+        precision = 2
+        for record in self:
+            if not float_is_zero(record.selling_price, precision_digits=precision):
+                if (float_compare(
+                        record.selling_price,
+                        0.9 * record.expected_price,
+                        precision_digits=precision,
```

Comment:
```suggestion
    for record in self:
        if not float_is_zero(record.selling_price, precision_digits=precision):
            if (float_compare(
                    record.selling_price,
                    0.9 * record.expected_price,
                    precision_digits=2,
```
no need to create variable for that